code/obj/item/beartrap.dm

```diff
@@ -40,31 +40,21 @@
 		else
 			..()
 
-	attack_self(mob/M as mob)
+	attack_self(mob/M)
 		if (!src.armed)
 			M.show_text("You start to arm the beartrap...", "blue")
 			var/datum/action/bar/icon/callback/action_bar = new /datum/action/bar/icon/callback(M, src, 2 SECONDS, /obj/item/beartrap/proc/arm,\
 			list(M), src.icon, src.icon_state, "[M] finishes arming [src]")
 			actions.start(action_bar, M)
 		return
 
-	Crossed(atom/movable/AM as mob|obj)
-		if ((ishuman(AM)) && (src.armed))
-			var/mob/living/carbon/H = AM
-			src.triggered(H)
-			H.visible_message("<span class='alert'><B>[H] steps on the bear trap!</B></span>",\
+	Crossed(atom/movable/AM)
+		if (!isghostcritter(AM) && isliving(AM) && !isintangible(AM) && src.armed)
```

Comment:
```suggestion
		if (src.armed &&  isliving(AM) && !(isintangible(AM) || !isghostcritter(AM)))
```
Order of evaluation in these things matters, since byond won't evaluate later statements if one already returned false. As such, it's good practice to order them by cheapest-to-evaluate first. This is a very nitpicky comment.